packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -49,250 +49,306 @@ const password = 'password123';
+          it('should create new vault, mnemonic and keychain', async () => {
+            await withController(
+              { cacheEncryptionKey },
+              async ({ controller, initialState, preferences, encryptor }) => {
+                const cleanKeyringController = new KeyringController(
+                  preferences,
+                  { cacheEncryptionKey, encryptor },
+                );
+                const initialSeedWord = await controller.exportSeedPhrase(
+                  password,
+                );
+                const currentState =
+                  await cleanKeyringController.createNewVaultAndKeychain(
+                    password,
+                  );
+                const currentSeedWord =
+                  await cleanKeyringController.exportSeedPhrase(password);
+                expect(initialSeedWord).toBeDefined();
+                expect(initialState).not.toBe(currentState);
+                expect(currentSeedWord).toBeDefined();
+                expect(initialSeedWord).not.toBe(currentSeedWord);
+                expect(
+                  isValidHexAddress(currentState.keyrings[0].accounts[0]),
+                ).toBe(true);
+              },
+            );
+          });
+        });
 
-    describe('when there is an existing vault', () => {
-      it('should return existing vault', async () => {
-        const initialSeedWord = await keyringController.exportSeedPhrase(
-          password,
-        );
-        const currentState = await keyringController.createNewVaultAndKeychain(
-          password,
-        );
-        const currentSeedWord = await keyringController.exportSeedPhrase(
-          password,
-        );
-        expect(initialSeedWord).toBeDefined();
-        expect(initialState).toBe(currentState);
-        expect(currentSeedWord).toBeDefined();
-        expect(initialSeedWord).toBe(currentSeedWord);
-      });
-    });
+        describe('when there is an existing vault', () => {
+          it('should return existing vault', async () => {
+            await withController(
+              { cacheEncryptionKey },
+              async ({ controller, initialState }) => {
+                const initialSeedWord = await controller.exportSeedPhrase(
+                  password,
+                );
+                const currentState = await controller.createNewVaultAndKeychain(
+                  password,
+                );
+                const currentSeedWord = await controller.exportSeedPhrase(
+                  password,
+                );
+                expect(initialSeedWord).toBeDefined();
+                expect(initialState).toBe(currentState);
+                expect(currentSeedWord).toBeDefined();
+                expect(initialSeedWord).toBe(currentSeedWord);
+              },
+            );
+          });
+        });
+
+        cacheEncryptionKey &&
+          it('should set encryptionKey and encryptionSalt in state', async () => {
```

Comment:
It seems like `createNewVaultAndKeychain` is not called in this test. Is that intentional?

Reply:
the `withController` function already calls `createNewVaultAndKeychain` to provide a functioning KeyringController instance to all the tests, so the controller we have in this case is already exactly what we need to make assertions

Reply:
Oh, I didn't see that. Gotcha.